l10n_nl_bsn/models/res_partner.py

```diff
@@ -0,0 +1,92 @@
+# Copyright 2016-2020 Onestein (<https://www.onestein.eu>)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+import logging
+
+from odoo import _, api, fields, models
+from odoo.osv import expression
+
+_logger = logging.getLogger(__name__)
+try:
+    from stdnum.nl import bsn
+except ImportError:
+    _logger.debug("Cannot `import stdnum.nl.bsn`.")
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    bsn_number = fields.Char(string="BSN", groups="hr.group_hr_user")
+
+    @api.onchange("bsn_number")
+    def onchange_bsn_number(self):
+        warning = {}
+        if self.bsn_number:
+            # properly format the entered BSN
+            self.bsn_number = bsn.format(self.bsn_number)
+
+            # check is valid, otherwise display a warning
+            warning = self._warn_bsn_invalid()
+
+            # search for another partner with the same BSN
+            args = [("bsn_number", "=", self.bsn_number), ("name", "!=", self.name)]
+
+            # refine search in case of multicompany setting
+            if self.company_id:
+                args += [("company_id", "=", self.company_id.id)]
```

Comment:
Should this not also includes a search for records where company is False?

Reply:
I assume that, in case `if self.company_id` is True, then a different partner having company == False would not conflict, so in this case not any warning is expected. The warning is expected only in case 2 partners have the same explicit company.

Reply:
Let us say we have two companies A and B. There is a partner 'Jan' with bsn '123' (just to lazy here to type a ful bsn) with no company set, this partner is available for company A and B. When we introduce a partner 'Piet' for company A with also bsn '123', both Jan and Piet are now available in company A with the same bsn number. I would count that as a conflict.

Reply:
That's debatable. My original intention was to warn only in case 2 partners have the same explicit company, which is the case that implies a conflict for sure. Instead, in case a user is mixing on purpose partner with companies and without companies with same BSN, that may be something wanted and not necessarily a conflict. In case someone wants to restrict the check, there are 2 options: 1. extend and customize the onchange method or 2. post a change request. At the moment for this migration I don't feel comfortable to modify the functional behavior of this module that turned 7 years old already.

Reply:
Let us leave it as it then for the moment.